pkg/dumper/dump.go

```diff
@@ -19,7 +25,65 @@ type Uploader interface {
 	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
 }
 
-func New(uploader Uploader, bucket string, logger *zap.Logger) S3DumpClient {
+type RoundRobinDumpClient struct {
```

Comment:
The `RoundRobinDumpClient` name makes me think that this is just another client like `LocalDumpClient` and `S3DumpClient`. Something like `RoundRobinDumpHandler` might be more intuitive.